csi/src/client.rs

```diff
@@ -16,11 +16,10 @@ use rpc::service::mayastor_client::MayastorClient;
 
 fn parse_share_protocol(pcol: Option<&str>) -> Result<i32, Status> {
     match pcol {
-        None => Ok(rpc::mayastor::ShareProtocol::None as i32),
+        Some("nbd") => Ok(rpc::mayastor::ShareProtocol::Nbd as i32),
```

Comment:
parse_share_protocol() looks like it is called only by create_replica() and share_replica(). 
Replicas use the back-end interface which use only iscsi and nvmf. Look at mayastor/src/replica.rs. 
nbd looks to like it should still be a invalid option.

Reply:
As was None, so no change there.

Reply:
Actually looking at the use cases
`$ find csi -name \*.rs -exec fgrep -Hn parse_share_protocol {} \;
csi/src/client.rs:17:fn parse_share_protocol(pcol: Option<&str>) -> Result<i32, Status> {
csi/src/client.rs:132:    let share = parse_share_protocol(matches.value_of("protocol"))?;
csi/src/client.rs:176:    let share = parse_share_protocol(matches.value_of("PROTOCOL"))?;
`
I think it the change is correct.
The underlying calls will fail if the share protocol is not supported by replicas.
The csi code should not make any assumption about the behaviour of replicas.